packages/web3-common/src/web3_base_provider.ts

```diff
@@ -23,7 +31,7 @@ const symbol = Symbol.for('web3/base-provider');
 
 // Provider interface compatible with EIP-1193
 // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md
-export abstract class Web3BaseProvider {
+export abstract class Web3BaseProvider<API extends Web3APISpec = EthExecutionAPI> {
```

Comment:
EthExecutionAPI must not be enforced at providers level, as in future we might integrate other APIs. 
Filtering transport for specific API functions will make it harder to use for future APIs.